src/libstore/build.cc

```diff
@@ -4461,7 +4471,29 @@ void Worker::waitForInput()
 
 unsigned int Worker::exitStatus()
 {
-    return timedOut ? 101 : (permanentFailure ? 100 : 1);
+    /*
+     * 1100100
+     *    ^^^^
+     *    |||`- timeout
+     *    ||`-- output hash mismatch
+     *    |`--- build failure
+     *    `---- not deterministic
+     */
+    unsigned int mask = 0;
+    bool buildFailure = permanentFailure || timedOut || hashMismatch;
+    if (buildFailure)
+        mask |= 0x04;  // 100
+    if (timedOut)
+        mask |= 0x01;  // 101
```

Comment:
This produces a value of `0x61 == 97` right?

Reply:
Would be nice to make this tested explicitly.

Reply:
0x04 + 0x01 + 0x60 (below) produces 101

Also I added a test ensuring this behavior remains true. Should be good to go?